DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/matmul/README.md

```diff
@@ -97,6 +112,41 @@ The key optimization techniques used are as follows:
 2. Using the `fpga_reg` attribute to insert additional pipelining registers, a crucial step in the implementation of the systolic array structure of this design, which allows data to be passed from one PE to the next. The use of these registers both in the systolic array and in various other parts of the design improves the overall achievable frequency.
 3. Using an efficient memory banking scheme to generate high performance hardware.
 
+### Comparison with a naïve approach
+
+Consider the following naïve implementation of a matrix multiplication kernel, where instead of implementing a systolic array, we unroll the loop computing the dot product. 
+```c++
+for (int row = 0; row < m; row++) {
+  for (int col = 0; col < p; col++) {
+    float dot_prod{0};
+#pragma unroll
+    for (int k = 0; k < n; k++) {
+      dot_prod = fpga_reg(dot_prod) + a_load[row][k] * b_load[col][k];
+    }
+    mm_result[row][col] = dot_prod;
+  }
+}
+```
+
+This achieves the same objective of increasing throughput, though this design is not as scalable as the systolic array approach, as shown in the results below.
+
+*Systolic array implementation*
+| Matrix size | PE array size | DSP usage | Avg. fMAX
+|:---         |:---           |:---       |:---
+| 64x64       | 8x8           | 64        | 452.55 MHz
+| 256x256     | 16x16         | 256       | 380.00 MHz
+| 256x256     | 32x16         | 512       | 346.72 MHz
```

Comment:
I would personally omit the two last rows as they cannot be compared with the ones in the naive implementation